include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,199 @@
+  using hpx::unwrapping;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+
+  if (m <= 1 || n == 0)
+    return;
+
+  const SizeType nr_reflector = mat_v.size().rows() - mb;
+
+  dlaf::matrix::Distribution dist_t({mb, nr_reflector}, {mb, mb});
+
+  constexpr std::size_t n_workspaces = 2;
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsV(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsW(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsW2(n_workspaces,
+                                                                         mat_c.distribution());
+  matrix::Panel<Coord::Row, T, Device::CPU> panelT(dist_t);
+
+  // Specific for V matrix layout where last column of tiles is empty
+  const SizeType num_panel_refls = mat_v.nrTiles().cols() - 1;
+  const SizeType last_tile_cols = mat_v.tileSize(GlobalTileIndex(0, m - 1)).cols();
+
+  for (SizeType k = num_panel_refls - 1; k >= 0; --k) {
+    bool is_last = (k == num_panel_refls - 1);
+    const SizeType t_last_cols = dist_t.tileSize(GlobalTileIndex{k, k}).cols();
+    const auto kt = k + 1;
+    const GlobalTileIndex v_start{kt, k};
+    auto kk = LocalTileIndex{k, k};
+
+    auto& panelV = panelsV.nextResource();
+    auto& panelW = panelsW.nextResource();
+    auto& panelW2 = panelsW2.nextResource();
+
+    panelV.setRangeStart(v_start);
+    panelW.setRangeStart(v_start);
+
+    if (is_last) {
+      panelT.setHeight(t_last_cols);
+      panelW2.setHeight(t_last_cols);
+      panelW.setWidth(last_tile_cols);
+      panelV.setWidth(last_tile_cols);
+    }
+
+    for (SizeType i = kt; i < mat_v.nrTiles().rows(); ++i) {
+      auto ik = LocalTileIndex{i, k};
+      if (i == kt) {
+        hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_v = mat_v.read(ik);
+        tile_v = splitTile(tile_v, {{0, 0},
+                                    {mat_v.distribution().tileSize(GlobalTileIndex(i, k)).rows(),
+                                     dist_t.tileSize(GlobalTileIndex(k, k)).cols()}});
+        copySingleTile(tile_v, panelV(ik));
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f, 1.f,
+                      panelV(ik));
+      }
+      else {
+        panelV.setTile(ik, mat_v.read(ik));
+      }
+
+      // Copy panelV into panelW (panelW will be modified in TRMM operation, while original tile of
+      // panelV will be needed in GEMM trailing matrix)
+      copySingleTile(panelV.read(ik), panelW(ik));

```

Comment:
I think it would be better to move this right before the trmm, such that it is clear why the copy is needed without the comment.